b2/console_tool.py

```diff
@@ -1040,6 +1042,53 @@ def _setup_logging(self, args, command, argv):
             logger.info('starting command [%s] with arguments: %s', command, argv)
 
 
+DATE_PATTERN = re.compile(r'^..., (\d\d) (...) (\d\d\d\d) (\d\d:\d\d\:\d\d) GMT$')
+
+MONTHS = ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']
+
+
+def clock_skew_hook(http_response):
+    """
+    Raises an exception if the clock in the server is too different from the
+    clock on the local host.
+
+    The Date header contains a string that looks like: "Fri, 16 Dec 2016 20:52:30 GMT".
+    The strptime function uses the current locale for month and day names like
+    "Dec" and "Fri", so we convert them to numbers ourselves.
+    """
+    # Make a string that uses month numbers instead of month names
+    server_date_str = http_response.headers['Date']
+    match = DATE_PATTERN.match(server_date_str)
+    if match is None:
+        raise DateFormatBad('date from server is: ' + server_date_str)
+    month_name = match.group(2)
+    if month_name not in MONTHS:
+        raise DateFormatBad('date from server is: ' + server_date_str)
+    month_number = MONTHS.index(month_name) + 1
+    server_numeric_date_str = '%s-%02d-%s %s GMT' % (
+        match.group(3), month_number, match.group(1), match.group(4)
+    )
+
+    # Convert the server time to a datetime object
+    server_time = datetime.datetime.strptime(server_numeric_date_str, '%Y-%m-%d %H:%M:%S %Z')
+
+    # Get the local time
+    local_time = datetime.datetime.utcnow()
+
+    # Check the difference.  The timedelta.total_seconds() method is not available
+    # in Python 2.6, so we'll compute it using the formula from the Python docs.
+    max_allowed = 10 * 60  # ten minutes, in seconds
+    skew = local_time - server_time
+    skew_seconds = int(
+        (skew.microseconds + (skew.seconds + skew.days * 24 * 3600) * 1000000) / 1000000
+    )
+    if max_allowed < abs(skew_seconds):
+        if skew_seconds < 0:
+            raise ClockSkew('local clock is %d seconds behind server' % (-skew_seconds,))
```

Comment:
the formatting of this message should not be here, but in `ClockSkew` class, the same as with other exceptions we have.